sympy/core/tests/test_power.py

```diff
@@ -507,3 +507,8 @@ def test_issue_17450():
 
 def test_issue_18190():
     assert sqrt(1 / tan(1 + I)) == 1 / sqrt(tan(1 + I))
+
+
+def test_issue_18509():
+    assert 2**oo / pi**oo == oo*pi**(-oo)
```

Comment:
This kind of test is tricky to get right because you can end up writing something where both sides of `==` evaluate to the wrong thing.

I think it would better to write
```
from sympy.core.expr import unchanged              # at the top of the file                                                                                  
assert unchanged(Mul, oo, 1/pi**oo)
```

Reply:
Okay I will upadte it.